nexus/operation.go

```diff
@@ -132,6 +133,8 @@ func (h *syncOperation[I, O]) Start(ctx context.Context, input I, options StartO
 type Service struct {
 	Name string
 
+	// Lock to protect operations map in case new services are added while workflows are running.
+	lock       sync.RWMutex
```

Comment:
Does this really need to be thread safe? Or can we copy the operations at registration time. Or are we intentionally supporting registering operations after a service is already registered?

Reply:
> It doesn't need to be thread safe but I worry that users will not read our docstrings and assume it is

That's not really how Go works. Everyone is supposed to assume not safe unless docs say otherwise. They will have issues with every other Go library including the standard library with those assumptions.

> We could disable late registration by adding a Seal() method to Service that is automatically called when the service is registered.

Not sure it should be public, but sure anything to prevent post-service-registration operation registration works for me

Reply:
It would need to be public for the Temporal SDK to call it.

Let's go ahead and do that then.
I think the bool doesn't need to be atomic, since it can be set under the lock and never gets unset. Let's make this thread safe, doesn't really hurt anything.

Reply:
> It would need to be public for the Temporal SDK to call it.

Ah, I figured they'd use the registry that's in here. Then I would not suggest `Seal()`, I would suggest `Clone()` that returns a copied instance.

Reply:
Hmm.. I think there's value in a `Seal()` API. It'd give a clear error if a user tries to register operations post handler creation. `Clone()` would be silent. Another (somewhat negligible) advantage to `Seal()` is that operation lookup doesn't require a lock or atomic variable if we do add thread safety to the registry.

Reply:
It's not a very common Go approach (nor is `Freeze` or terms like that). Usually you mutate something and you pass it. You might not expect the thing you pass it to to also mutate (by calling `Seal`).

Reply:
So maybe let's just keep this not thread safe and let users deal with the implications of that if it's a more common Go approach.
Could be good to stress that the fact that it's not thread safe also means that users should not try to register operations after the service has been registered.

Reply:
Works for me. And really, the "Seal" thing works for me too if you'd like to actually prevent post-servce-registration operation registration, though by-copy/clone type of approaches are more common.